packages/cloud-cognitive/src/components/SidePanel/SidePanel.js

```diff
@@ -505,23 +507,25 @@ export let SidePanel = React.forwardRef(
             {actionToolbarButtons.map((action) => (
               <Button
                 key={action.label}
-                kind={action.leading ? action.kind : 'ghost'}
+                kind={action.kind || 'ghost'}
                 size="small"
-                disabled={false}
                 renderIcon={action.icon}
                 iconDescription={action.label}
                 tooltipPosition="bottom"
                 tooltipAlignment="center"
+                hasIconOnly={!action.leading}
+                disabled={action.disabled}
                 className={cx([
                   `${blockClass}__action-toolbar-button`,
+                  action.className,
                   {
                     [`${blockClass}__action-toolbar-icon-only-button`]:
-                      action.icon,
+                      action.icon && !action.leading,
                     [`${blockClass}__action-toolbar-leading-button`]:
-                      !action.icon,
+                      action.leading,
                   },
                 ])}
-                onClick={() => action.onActionToolbarButtonClick()}>
+                onClick={() => action.onClick()}>
```

Comment:
This makes sense for both consistency and predictability — will it break existing consumers?

One other thought, and I likely need a history lesson on this, but is there a rationale for preventing the event parameter to be passed to the callback?

Reply:
Updated @SimonFinney

Reply:
<img width="1436" alt="Screen Shot 2021-08-23 at 6 07 40 PM" src="https://user-images.githubusercontent.com/10215203/130527374-92cda2e1-fb6c-43b4-a71d-c30f45aa9b7e.png">

Used the `deprecateProp` helper that will give a console warning if `onActionToolbarButtonClick` is used.